app/recipe/tests/test_recipe_api.py

```diff
@@ -0,0 +1,208 @@
+from core.models import Recipe, Ingredient
+from django.test import TestCase
+from django.urls import reverse
+from recipe.serializers import RecipeSerializer
+from rest_framework import status
+from rest_framework.test import APIClient
+
+RECIPES_URL = reverse('recipe:recipe-list')
+
+
+def recipe_url(recipe_id):
+    """Return URL for a recipe"""
+
+    return reverse('recipe:recipe-detail', args=[recipe_id])
+
+
+def sample_recipe(params={}):
```

Comment:
FYI Not part of the exercise, but we use 
https://factoryboy.readthedocs.io/en/latest/orms.html#django
to generate dummy data.

Reply:
Cheers good to know, was just following what the vid showed us to do.